
# 수행평가 문제 3 - 1
- timestamp : 1970년 1월 1일 0시 0분 0초를 0으로 설정하고 1초마다 1씩 증가
- 문제 출처 : [링크](https://moon9342.github.io/data-science-lecture-python-pandas-dataframe-exercise)
---
1. 사용자가 평가한 모든 영화의 전체 평균 평점
2. 각 사용자별 평균 평점
3. 각 영화별 평균 평점
4. 평균 평점이 가장 높은 영화의 제목(동률이 있을 경우 모두 출력)
5. Comedy영화 중 가장 평균 평점이 낮은 영화의 제목
6. 2015년도에 평가된 모든 Romance 영화의 평균 평점은?

In [ ]:
import numpy as np
import pandas as pd
from datetime import datetime

# 데이터 가져오기
ratings = pd.read_csv('./data/ml-latest-small/ratings.csv')
movies = pd.read_csv('./data/ml-latest-small/movies.csv')

In [ ]:
# 데이터 확인
# 1. nan값 있는지 없는지
print(ratings.isnull().sum())
print(movies.isnull().sum())

# 데이터 개요
# 사용자가 몇명인지?
print(ratings.groupby(ratings['userId']).count())
# 리뷰가 기록된 영화가 몇개인지?
print(ratings.groupby(ratings['movieId']).count())
# 영화 데이터 개수
print(movies.count())

- 데이터 전처리
1. timestamp 처리 
    - 참고문헌(datetime.fromtimestamp()) [링크](https://docs.python.org/3/library/datetime.html?highlight=timestamp#datetime.datetime.fromtimestamp)
2. genres 처리
    - 참고문헌(DataFrame.apply()) [링크](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.apply.html#pandas.DataFrame.apply)

In [ ]:
# timestamp처리
ratings['timestamp'] = ratings['timestamp'].apply(datetime.fromtimestamp)

In [ ]:
# genres 처리
# 찍어보면 str로 이루어짐
print(movies['genres'][0])
print(type(movies['genres'][0]))
# 처리방법 1. split해서 리스트로 만든다?

# 일단 리스트로 바꿔보자
movies['genres'] = movies['genres'].apply(lambda x : x.split('|'))

In [ ]:
# 전처리 끝난 DataFrame 하나로 연결
# 평점이 없는 영화가 존재하므로 right outer join
data = pd.merge(ratings, movies, on='movieId', how='right')
display(data)

## 1. 사용자가 평가한 모든 영화의 전체 평균 평점

In [ ]:
print(data['rating'].mean())

## 2. 각 사용자별 평균 평점

In [ ]:
print(data.groupby(data['userId'])['rating'].mean())

## 3. 각 영화별 평균 평점

In [ ]:
print(data.groupby(data['movieId'])['rating'].mean())

## 4. 평균 평점이 가장 높은 영화의 제목(동률이 있을 경우 모두 출력)

In [ ]:
# 일단 최고 평점은 5.0
score = data.groupby(data['title'])['rating'].mean()
print(score.max())
# 혹시 동률있나 확인
display(score[score == 5.0] )
# 5.0인 영화 296개

5. Comedy영화 중 가장 평균 평점이 낮은 영화의 제목

In [ ]:
tmp = []
for a in data['genres']:
    tmp.append('Comedy' in a)
# Comedy영화 boolean mask
tmp_arr = np.array(tmp)

In [ ]:
score = data[tmp_arr].groupby(data[tmp_arr]['title'])['rating'].mean()
# 최저 평점 0.5
print(score.min())
# 0.5인 Comedy 영화
print(score[score == 0.5])

## 6. 2015년도에 평가된 모든 Romance 영화의 평균 평점은?

In [ ]:
tmp = []
for a in data['genres']:
    tmp.append('Romance' in a)
# Romance 영화 boolean mask
tmp_arr = np.array(tmp)

tmp = []
for a in data['timestamp']:
    tmp.append(a.year == 2015)
# Romance 영화 boolean mask & 2015년도에 평가된 리뷰 boolean mask
tmp_arr = np.array(tmp) & tmp_arr

print(data[tmp_arr]['rating'].mean())

---
# 수행평가 문제 3 - 2
---

1. displ(배기량)이 4 이하인 자동차와 5 이상인 자동차 중 어떤 자동차의 hwy(고속도로 연비)가 평균적으로 더 높은지 확인하세요.
2. 자동차 제조 회사에 따라 도시 연비가 다른지 알아보려고 한다. "audi"와 "toyota" 중 어느 manufacturer(제조회사)의 cty(도시 연비)가 평균적으로 더 높은지 확인하세요.
3. "chevrolet", "ford", "honda" 자동차의 고속도로 연비 평균을 알아보려고 한다. 이 회사들의 데이터를 추출한 후 hwy(고속도로 연비) 평균을 구하세요.
4. "audi"에서 생산한 자동차 중에 어떤 자동차 모델의 hwy(고속도로 연비)가 높은지 알아보려고 한다. "audi"에서 생산한 자동차 중 hwy가 1~5위에 해당하는 자동차의 데이터를 출력하세요.
5. mpg 데이터는 연비를 나타내는 변수가 2개입니다. 두 변수를 각각 활용하는 대신 하나의 통합 연비 변수를 만들어 사용하려 합니다. 평균 연비 변수는 두 연비(고속도로와 도시)의 평균을 이용합니다. 회사별로 "suv" 자동차의 평균 연비를 구한후 내림차순으로 정렬한 후 1~5위까지 데이터를 출력하세요.
6. mpg 데이터의 class는 "suv", "compact" 등 자동차의 특징에 따라 일곱 종류로 분류한 변수입니다. 어떤 차종의 도시 연비가 높은지 비교하려 합니다. class별 cty 평균을 구하고 cty 평균이 높은 순으로 정렬해 출력하세요.
7. 어떤 회사 자동차의 hwy(고속도로 연비)가 가장 높은지 알아보려 합니다. hwy(고속도로 연비) 평균이 가장 높은 회사 세 곳을 출력하세요.
8. 어떤 회사에서 "compact" 차종을 가장 많이 생산하는지 알아보려고 합니다. 각 회사별 "compact" 차종 수를 내림차순으로 정렬해 출력하세요.
- 저번에 class로 해결한거 DataFrame으로 해결하기

In [ ]:
import numpy as np
import pandas as pd

# 저번에 사용한 파일 csv로 바꿔서 불러오기
data = pd.read_csv('./data/mpg.csv')
display(data)

## 1. displ(배기량)이 4 이하인 자동차와 5 이상인 자동차 중 어떤 자동차의 hwy(고속도로 연비)가 평균적으로 더 높은지 확인하세요.


In [ ]:
print('배기량 4이하인 자동차의 고속도로 연비 : {}'.format(data[data['displ'] <= 4]['hwy'].mean()))
print('배기량 5이상인 자동차의 고속도로 연비 : {}'.format(data[data['displ'] >= 5]['hwy'].mean()))
# 4이하인 자동차가 더 높다

## 2. 자동차 제조 회사에 따라 도시 연비가 다른지 알아보려고 한다.  "audi"와 "toyota" 중 어느 manufacturer(제조회사)의 cty(도시 연비)가 평균적으로 더 높은지 확인하세요.


In [ ]:
print('audi 자동차의 도시 연비 : {}'.format(data[data['manufacturer']=='audi']['cty'].mean()))
print('toyota 자동차의 도시 연비 : {}'.format(data[data['manufacturer']=='toyota']['cty'].mean()))
# toyota가 더 높다

## 3. "chevrolet", "ford", "honda" 자동차의 고속도로 연비 평균을 알아보려고 한다. 이 회사들의 데이터를 추출한 후 hwy(고속도로 연비) 평균을 구하세요.


In [ ]:
data_3 = data[(data['manufacturer']=='chevrolet')
              |(data['manufacturer']=='ford')
              |(data['manufacturer']=='honda')]
print(data_3.groupby(data_3['manufacturer']).mean()['cty'])

## 4. "audi"에서 생산한 자동차 중에 어떤 자동차 모델의 hwy(고속도로 연비)가 높은지 알아보려고 한다. "audi"에서 생산한 자동차 중 hwy가 1~5위에 해당하는 자동차의 데이터를 출력하세요.


In [ ]:
display(data[data['manufacturer']=='audi'].sort_values(['hwy'], ascending=False).head(5))

## 5. mpg 데이터는 연비를 나타내는 변수가 2개입니다. 두 변수를 각각 활용하는 대신 하나의 통합 연비 변수를 만들어 사용하려 합니다. 평균 연비 변수는 두 연비(고속도로와 도시)의 평균을 이용합니다. 회사별로 "suv" 자동차의 평균 연비를 구한후 내림차순으로 정렬한 후 1~5위까지 데이터를 출력하세요.


In [ ]:
# 새로운 colums 만들기
data['mean'] = (data['cty'] + data['hwy'])/2
display(data)
# 회사별 suv 연비 1~5 순위 출력
for a in data['manufacturer'].unique():
    print("{} 회사의 suv 자동차 연비 1~5위".format(a))
    display(data[(data['class']=='suv')&(data['manufacturer']==a)].sort_values(['mean'], ascending=False).head(5))
    
# suv 없는 회사도 출력해야 하나?

In [ ]:
# suv 없는 회사를 위한 추가 작업
data['mean'] = (data['cty'] + data['hwy'])/2

# 회사별 suv 연비 1~5 순위 출력
for a in data['manufacturer'].unique():
    if data[(data['class']=='suv')&(data['manufacturer']==a)].count()[0] == 0:
        print('{} 회사의 suv 자동차 자료 없음'.format(a))
    else:
        print("{} 회사의 suv 자동차 연비 1~5위".format(a))
        display(data[(data['class']=='suv')&(data['manufacturer']==a)].sort_values(['mean'], ascending=False).head(5))

## 6. mpg 데이터의 class는 "suv", "compact" 등 자동차의 특징에 따라 일곱 종류로 분류한 변수입니다. 어떤 차종의 도시 연비가 높은지 비교하려 합니다. class별 cty 평균을 구하고 cty 평균이 높은 순으로 정렬해 출력하세요.


In [ ]:
display(data.groupby(data['class']).mean().sort_values(['cty'], ascending=False))

## 7. 어떤 회사 자동차의 hwy(고속도로 연비)가 가장 높은지 알아보려 합니다. hwy(고속도로 연비) 평균이 가장 높은 회사 세 곳을 출력하세요.


In [ ]:
display(data.groupby(data['manufacturer']).mean().sort_values(['hwy'], ascending=False).head(3))

## 8. 어떤 회사에서 "compact" 차종을 가장 많이 생산하는지 알아보려고 합니다. 각 회사별 "compact" 차종 수를 내림차순으로 정렬해 출력하세요.

In [ ]:
display(data[data['class']=='compact'].groupby(data['manufacturer']).count().sort_values(['class'], ascending=False)['class'])

---
# 수행평가 문제 3 - 3
---
1. 성별에 따른 월급 차이 
    - 과거에 비해 여성의 사회 진출이 활발하지만 직장에서의 위상에서는 여전히 차별이 존재하고 있는것이 사실. 실제로 그러한지 월급의 차이를 이용하여 사실을 확인해보자.

2. 나이와 월급의 관계 
    - 평균적으로 몇 살 때 월급을 가장 많이 받을까? 또 그때의 평균 월급은 얼마인가?

3. 연령대에 따른 월급 차이 
    - [30세 미만을 초년(young), 30~59세 : 중년(middle), 60세 이상 : 노년(old)] 
    - 위의 범주로 연령대에 따른 월급의 차이를 알아보자

4. 연령대 및 성별 월급 차이 
    - 성별 월급 차이는 연령대에 따라 다른 양상을 보일 수 있습니다. 성별 월급 차이가 연령대에 따라 다른지 분석해보자. 기존에는 3그룹(초년,중년,노년)이었지만 이젠 6그룹으로 그룹핑을 해야 한다.(초년남성,초년여성,..)

5. 직업별 월급 차이
    - 어떤 직업이 월급을 가장 많이 받을까? 직업별 월급을 분석해 보자. 직업코드는 제공된 Koweps_Codebook.xlsx을 이용하면 편하게 코드값을 이용 할 수 있습니다.

6. 성별 직업 빈도
    - 성별로 어떤 직업이 가장 많을까?

7. 종교 유무에 따른 이혼율
    - 종교가 있는 사람들이 이혼을 덜 할까??

8. 지역별 연령대 비율
    - 노년층이 많은 지역은 어디일까?

In [1]:
# 한국복지패널데이터를 이용한 데이터 분석 연습

# 데이터 준비

# 제공된 파일의 압축을 풀면 파일이 2개 있습니다. 
# Koweps_hpc10_2015_beta1.sav : SPSS 데이터 파일 (128M)
# Koweps_Codebook.xlsx : 우리가 사용할 column에 대한 설명
# 총 7개의 column만을 이용합니다. 


# 데이터 파일인 Koweps_hpc10_2015_beta1.sav 파일은 
# 상용 통계분석 소프트웨어인 SPSS 전용파일입니다. 

# 해당 데이터 파일은 SPSS 전용파일이기 때문에 pandas에서 사용하기 
# 위해서 다음의 코드를 이용해서 DataFrame으로 변환해야 합니다.


# pip install savReaderWriter 
import numpy as np
import pandas as pd
import savReaderWriter

with savReaderWriter.SavReader('./data/Koweps/Koweps_hpc10_2015_beta1.sav', ioUtf8 = True) as reader:
    df = pd.DataFrame(reader.all(), columns = [s for s in reader.header])

print(df.shape)      # # (16664, 957)
df['h10_g3'].head()  # 성별
df['h10_g4'].head()  # 태어난 연도

(16664, 957)


0    1936.0
1    1945.0
2    1948.0
3    1942.0
4    1923.0
Name: h10_g4, dtype: float64

In [2]:
display(df[['h10_g3','h10_g4','h10_g10','h10_g11','h10_eco9','p1002_8aq1','h10_reg7']])
display(df[['h10_g3','h10_g4','h10_g10','h10_g11','h10_eco9','p1002_8aq1','h10_reg7']].describe())

h10_g3  h10_g4  h10_g10  h10_g11  h10_eco9  p1002_8aq1  h10_reg7
0         2.0  1936.0      2.0      2.0       NaN         NaN       1.0
1         2.0  1945.0      2.0      2.0       NaN         NaN       1.0
2         1.0  1948.0      2.0      2.0     942.0       120.0       1.0
3         1.0  1942.0      3.0      1.0     762.0       200.0       1.0
4         2.0  1923.0      2.0      1.0       NaN         NaN       1.0
...       ...     ...      ...      ...       ...         ...       ...
16659     2.0  1967.0      1.0      1.0       NaN         NaN       5.0
16660     2.0  1992.0      5.0      1.0     314.0       302.5       5.0
16661     1.0  1995.0      5.0      1.0       NaN         NaN       5.0
16662     2.0  1998.0      0.0      1.0       NaN         NaN       5.0
16663     1.0  2001.0      0.0      1.0       NaN         NaN       5.0

[16664 rows x 7 columns]

h10_g3        h10_g4       h10_g10       h10_g11     h10_eco9  \
count  16664.000000  16664.000000  16664.000000  16664.000000  7529.000000   
mean       1.545247   1967.568711      1.647744      1.517103   591.243724   
std        0.497963     24.179743      1.571827      0.499722   255.793317   
min        1.000000   1907.000000      0.000000      1.000000   111.000000   
25%        1.000000   1946.000000      1.000000      1.000000   314.000000   
50%        2.000000   1966.000000      1.000000      2.000000   611.000000   
75%        2.000000   1988.000000      2.000000      2.000000   863.000000   
max        2.000000   2014.000000      6.000000      2.000000  1012.000000   

        p1002_8aq1      h10_reg7  
count  4634.000000  16664.000000  
mean    241.619016      3.704873  
std     184.423869      2.069207  
min       0.000000      1.000000  
25%     122.000000      2.000000  
50%     192.500000      3.000000  
75%     316.600000      6.000000  
max    2400.000000      7.000000

In [3]:
# EDA
# 성별 데이터 확인
print('성별 미응답자 : {}'.format(df[df['h10_g3']==9].count()['h10_g3']))
print('성별 누락 데이터 : {}'.format(df['h10_g3'].isnull().sum()))
# 출생연도 데이터 확인
print('출생연도 미응답자 : {}'.format(df[df['h10_g4']==9999].count()['h10_g4']))
print('출생연도 누락 데이터 : {}'.format(df['h10_g4'].isnull().sum()))
# 혼인상태 데이터 확인
print('혼인상태 미응답자 : {}'.format(df[df['h10_g10']==9].count()['h10_g10']))
print('혼인상태 누락 데이터 : {}'.format(df['h10_g10'].isnull().sum()))
# 종교 데이터 확인
print('종교 미응답자 : {}'.format(df[df['h10_g11']==9].count()['h10_g11']))
print('종교 누락 데이터 : {}'.format(df['h10_g11'].isnull().sum()))
# 직종 데이터 확인
print('직종 미응답자 : {}'.format(df[df['h10_eco9']==9999].count()['h10_eco9']))
print('직종 누락 데이터 : {}'.format(df['h10_eco9'].isnull().sum()))
# 월 평균 임금 데이터 확인
print('월 평균 임금 미응답자 : {}'.format(df[df['p1002_8aq1']==9999].count()['p1002_8aq1']))
print('월 평금 인금 누락 데이터 : {}'.format(df['p1002_8aq1'].isnull().sum()))
# 지역 데이터 확인
print('지역 누락 데이터 : {}'.format(df['h10_reg7'].isnull().sum()))

성별 미응답자 : 0
성별 누락 데이터 : 0
출생연도 미응답자 : 0
출생연도 누락 데이터 : 0
혼인상태 미응답자 : 0
혼인상태 누락 데이터 : 0
종교 미응답자 : 0
종교 누락 데이터 : 0
직종 미응답자 : 0
직종 누락 데이터 : 9135
월 평균 임금 미응답자 : 0
월 평금 인금 누락 데이터 : 12030
지역 누락 데이터 : 0


- 직종과 임금을 제외한 다른 자료는 미응답과 누락데이터는 없음
- 직종과 임금의 경우 일을 하지 않는 경우에 해당하므로 이를 고려하여 문제를 해결해야함

## 1. 성별에 따른 월급 차이 
- 과거에 비해 여성의 사회 진출이 활발하지만 직장에서의 위상에서는 여전히 차별이 존재하고 있는것이 사실. 실제로 그러한지 월급의 차이를 이용하여 사실을 확인해보자.

In [ ]:
# 여기엔 미 응답자도 포함?
display(df.groupby(df['h10_g3']).mean()['p1002_8aq1'])

## 2. 나이와 월급의 관계 
- 평균적으로 몇 살 때 월급을 가장 많이 받을까? 또 그때의 평균 월급은 얼마인가?

In [ ]:
df_2 = df.groupby(df['h10_g4']).mean().sort_values(['p1002_8aq1'], ascending=False)['p1002_8aq1']
print('가장 많은 월급을 받는 나이 : {}, 평균 월급 : {}'.format(2021-(df_2.index[0]+1), df_2[df_2.index[0]]))

## 3. 연령대에 따른 월급 차이 
- [30세 미만을 초년(young), 30~59세 : 중년(middle), 60세 이상 : 노년(old)] 
- 위의 범주로 연령대에 따른 월급의 차이를 알아보자

In [ ]:
df_4 = df[['h10_g4','p1002_8aq1']].copy()
df_4['age'] = 2021-(df_4['h10_g4']+1)

In [ ]:
# age_range = 0 초년, = 1 중년, = 2 노년
df_4.loc[df_4['age'] < 30,'age_range'] = 0
df_4.loc[(df_4['age'] >= 30)&(df_4['age'] < 60),'age_range'] = 1
df_4.loc[df_4['age'] >= 60,'age_range'] = 2

In [ ]:
result = df_4.groupby(df_4['age_range']).mean()['p1002_8aq1']
print('초년의 월급 : {}'.format(result[0.0]))
print('중년의 월급 : {}'.format(result[1.0]))
print('노년의 월급 : {}'.format(result[2.0]))

## 4. 연령대 및 성별 월급 차이 
- 성별 월급 차이는 연령대에 따라 다른 양상을 보일 수 있습니다. 성별 월급 차이가 연령대에 따라 다른지 분석해보자. 기존에는 3그룹(초년,중년,노년)이었지만 이젠 6그룹으로 그룹핑을 해야 한다.(초년남성,초년여성,..)

In [ ]:
df_4.loc[:,'sex'] = df['h10_g3']
result = df_4.groupby([df_4['age_range'],df_4['sex']])['p1002_8aq1']
print('초년 남성의 월급 : {}'.format(result.get_group((0,1)).mean()))
print('초년 여성의 월급 : {}'.format(result.get_group((0,2)).mean()))
print('중년 남성의 월급 : {}'.format(result.get_group((1,1)).mean()))
print('중년 여성의 월급 : {}'.format(result.get_group((1,2)).mean()))
print('노년 남성의 월급 : {}'.format(result.get_group((2,1)).mean()))
print('노년 여성의 월급 : {}'.format(result.get_group((2,2)).mean()))

## 5. 직업별 월급 차이
- 어떤 직업이 월급을 가장 많이 받을까? 직업별 월급을 분석해 보자. 직업코드는 제공된 Koweps_Codebook.xlsx을 이용하면 편하게 코드값을 이용 할 수 있습니다.

In [ ]:
result = df.groupby(df['h10_eco9']).mean().sort_values(['p1002_8aq1'], ascending=False)['p1002_8aq1']
print(result)
print('가장 월급이 높은 직업은 : {}(금속 재료 공학 기술자 및 시험원)'.format(result.index[0]))

## 6. 성별 직업 빈도
- 성별로 어떤 직업이 가장 많을까?

In [9]:
# 필요한 데이터만 추출
# 직업이 없는 사람은 0으로 표기
df_6 = df[['h10_g3','h10_eco9']].fillna(0).copy()
display(df_6)

h10_g3  h10_eco9
0         2.0       0.0
1         2.0       0.0
2         1.0     942.0
3         1.0     762.0
4         2.0       0.0
...       ...       ...
16659     2.0       0.0
16660     2.0     314.0
16661     1.0       0.0
16662     2.0       0.0
16663     1.0       0.0

[16664 rows x 2 columns]

In [33]:
# 남성의 경우
df_6.loc[df_6['h10_g3'] == 1].groupby(df_6['h10_eco9']).count().sort_values(['h10_g3'], ascending=False)['h10_g3']

h10_eco9
0.0       3551
611.0      640
873.0      251
312.0      213
510.0      141
          ... 
213.0        2
876.0        2
1011.0       2
871.0        2
289.0        1
Name: h10_g3, Length: 142, dtype: int64

In [35]:
# 여성의 경우
df_6.loc[df_6['h10_g3'] == 2].groupby(df_6['h10_eco9']).count().sort_values(['h10_g3'], ascending=False)['h10_g3']

h10_eco9
0.0       5584
611.0      680
941.0      228
521.0      221
930.0      185
          ... 
273.0        1
242.0        1
236.0        1
223.0        1
1012.0       1
Name: h10_g3, Length: 105, dtype: int64

- 모든 성별에서 직업이 없는 사람이 가장 많고, 그 다음으로 많은 직업은 611 : 작물재배종사자이다. 

## 7. 종교 유무에 따른 이혼율
- 종교가 있는 사람들이 이혼을 덜 할까??

In [ ]:
df_7 = df[['h10_g10','h10_g11']].copy()
# 종교가 없는 사람 수
print((df_7['h10_g11']==2).sum())
# 종교가 없는데 이혼
print(df_7[df_7['h10_g11']==2].loc[df_7['h10_g10']==3].count()[0])
# 종교가 없는 사람의 이혼율
print('종교가 없는 사람의 이혼율 : {}'.format(df_7[df_7['h10_g11']==2].loc[df_7['h10_g10']==3].count()[0]/(df_7['h10_g11']==2).sum()))
# 종교가 있는 사람 수
print((df_7['h10_g11']==1).sum())
# 종교가 있는데 이혼
print(df_7[df_7['h10_g11']==1].loc[df_7['h10_g10']==3].count()[0])
# 종교가 있는 사람의 이혼율
print('종교가 없는 사람의 이혼율 : {}'.format(df_7[df_7['h10_g11']==1].loc[df_7['h10_g10']==3].count()[0]/(df_7['h10_g11']==1).sum()))

## 8. 지역별 연령대 비율
- 노년층이 많은 지역은 어디일까?

In [ ]:
df_8 = df[['h10_g4','h10_reg7']].copy()
# 출생연도 - 나이 변환
df_8['age'] = 2021-(df_8['h10_g4']+1)
# age_range에 노년 여부 추가
df_8.loc[df_8['age'] < 60,'age_range'] = 0
df_8.loc[df_8['age'] >= 60,'age_range'] = 1
# 지역 구분 문자열로 변경
reg_list=['서울','수도권(인천/경기)',
          '부산/경남/울산','대구/경북',
          '대전/충남','강원/충북',
          '광주/전남/전북/제주도']

for a in range(len(reg_list)):
    df_8.loc[df['h10_reg7']==(a+1),'h10_reg7'] = reg_list[a]

display(df_8.groupby(df_8['h10_reg7']).sum().sort_values(['age_range'], ascending=False)['age_range'])